src/main/java/com/gluonhq/gradle/tasks/ConfigBuild.java

```diff
@@ -61,74 +62,74 @@
+        Triplet targetTriplet;
         String target = clientExtension.getTarget().toLowerCase(Locale.ROOT);
         switch (target) {
-            case Constants.TARGET_HOST:
-                if (osname.contains("mac")) {
-                    targetTriplet = new TargetTriplet(Constants.AMD64_ARCH, Constants.HOST_MAC, Constants.TARGET_MAC);
-                } else if (osname.contains("nux")) {
-                    targetTriplet = new TargetTriplet(Constants.AMD64_ARCH, Constants.HOST_LINUX, Constants.TARGET_LINUX);
-                }
+            case Constants.PROFILE_HOST:
+                targetTriplet = Triplet.fromCurrentOS();
+                break;
+            case Constants.PROFILE_IOS:
+                targetTriplet = new Triplet(Constants.Profile.IOS);
                 break;
-            case Constants.TARGET_IOS:
-                targetTriplet = new TargetTriplet(Constants.ARM64_ARCH, Constants.HOST_MAC, Constants.TARGET_IOS);
+            case Constants.PROFILE_IOS_SIM:
+                targetTriplet = new Triplet(Constants.Profile.IOS_SIM);
                 break;
-            case Constants.TARGET_IOS_SIM:
-                targetTriplet = new TargetTriplet(Constants.AMD64_ARCH, Constants.HOST_MAC, Constants.TARGET_IOS);
+            case Constants.PROFILE_ANDROID:
+                targetTriplet = new Triplet(Constants.Profile.ANDROID);
+                break;
+            case Constants.PROFILE_LINUX_AARCH64:
+                targetTriplet = new Triplet(Constants.Profile.LINUX_AARCH64);
                 break;
             default:
                 throw new RuntimeException("No valid target found for " + target);
         }
         clientConfig.setTarget(targetTriplet);
 
-        clientConfig.setBackend(clientExtension.getBackend().toLowerCase(Locale.ROOT));
         clientConfig.setBundlesList(clientExtension.getBundlesList());
         clientConfig.setResourcesList(clientExtension.getResourcesList());
-        clientConfig.setDelayInitList(clientExtension.getDelayInitList());
         clientConfig.setJniList(clientExtension.getJniList());
+        clientConfig.setCompilerArgs(clientExtension.getCompilerArgs());
         clientConfig.setReflectionList(clientExtension.getReflectionList());
-        clientConfig.setRuntimeArgsList(clientExtension.getRuntimeArgsList());
-        clientConfig.setReleaseSymbolsList(clientExtension.getReleaseSymbolsList());
-
-        clientConfig.setMainClassName((String) project.getProperties().get("mainClassName"));
+        clientConfig.setAppId(project.getGroup() + "." + project.getName());
         clientConfig.setAppName(project.getName());
 
-        List<Path> classPath = getClassPathFromSourceSets();
-        clientConfig.setUseJavaFX(classPath.stream().anyMatch(f -> f.getFileName().toString().contains("javafx")));
-        clientConfig.setGraalLibsUserPath(clientExtension.getGraalLibsPath());
+        clientConfig.setGraalPath(Path.of(getGraalHome().get()));
```

Comment:
No exception or message is printed if the user hasn't set `GRAALVM_HOME`. In case of the maven plugin, [an exception is thrown](https://github.com/gluonhq/client-maven-plugin/blob/cf83a644f0cc3ec1c0b71f1784148e33413f4efd/src/main/java/com/gluonhq/NativeBaseMojo.java#L138) so that the end user is aware and can fix it.